force-app/main/default/classes/OppClosedWonDeletionValidation.cls

```diff
@@ -0,0 +1,45 @@
+public with sharing class OppClosedWonDeletionValidation {
+    public static void oppsToBeDeleted (List<Opportunity> deletedOpps) {
+        // First build a Set of Account Id's and a set of Opportunity Id's with each Opportunity where the StageName == 'Closed Won'
+        Set<Id> closedWonOppIds = new Set<Id>();
+        for (Opportunity opp : deletedOpps) {
+            if (opp.StageName == 'Closed Won') {
+                closedWonOppIds.add(opp.Id);
+            }
+        }
+
+         Set<Id> accountIdsOfClosedWonOpps = new Set<Id>();
+        for (Opportunity opp : [SELECT AccountId FROM Opportunity WHERE Id IN :closedWonOppIds]) {
+            accountIdsOfClosedWonOpps.add(opp.AccountId);
+        }
+        // Then I queried the Accounts from the Set of Account Id's where the Industry = 'Banking' and stored them in a Map<Id, Account>
+        Map<Id, Account> accountIdToAccountMap = new Map<Id, Account>([
+            SELECT Id 
+            FROM Account 
+            WHERE Id IN :accountIdsOfClosedWonOpps 
+                AND Industry = 'Banking'
+        ]);
+        // Then I looped back through the original list of Opportunities and if it was contained in my Set of opportunity Id's and the Map contained opp.AccountId then I threw an error
+        for (Opportunity opp : deletedOpps) {
+            if (closedWonOppIds.contains(opp.Id) && accountIdToAccountMap.containsKey(opp.AccountId)) {
+                opp.addError('Cannot delete closed opportunity for a banking account that is won');
+            }
+        }
+    }
+}
```

Comment:
Good job solving this. There were a lot of parts. I think you made it harder on yourself that it had to be. But that is part of learning.  My solution is below  if you are interested. 

```
 private void validateDeletedRecords(){
        Set<Id> accountIds = new Set<Id>();
        for(Opportunity opportunity : (List<Opportunity>)Trigger.old){
            accountIds.add(opportunity.AccountId);
        }
        Map<Id, Account> bankingAccountsById  = new Map<Id, Account>([SELECT Id, Industry
        FROM Account
        WHERE Id IN :accountIds
        AND Industry = 'Banking']);

        for(Opportunity opportunity : (List<Opportunity>)Trigger.old){
            if(bankingAccountsById.containsKey(opportunity.AccountId) && opportunity.StageName == 'Closed Won'){
                opportunity.addError('Cannot delete closed opportunity for a banking account that is won');
            }
        }
    }
    ```

Reply:
This makes more sense and the code you suggested looks clear and concise. But in the iterator as the Trigger.old is already a list of opportunities, do we need to cast that again to List<Opportunity>? Can I know the reason behind its inclusion? Thanks.